firedantic/tests/tests_async/test_model.py

```diff
@@ -534,4 +536,116 @@ async def test_save_with_exclude_unset(configure_db) -> None:
+            return p
+
+    transaction = CONFIGURATIONS["db"].transaction()
+    p = await create_in_transaction(transaction, "test@example.com")
+    assert isinstance(p, Profile)
+    assert p.id
+
+    transaction2 = CONFIGURATIONS["db"].transaction()
+    with pytest.raises(ValueError) as excinfo:
+        await create_in_transaction(transaction2, "test@example.com")
+    assert str(excinfo.value) == "Profile already exists with email: test@example.com"
+
+
+@pytest.mark.asyncio
+async def test_delete_in_transaction(configure_db) -> None:
+    """Test deleting a model in a transaction."""
+
+    @async_transactional  # type: ignore
+    async def delete_in_transaction(
+        transaction: AsyncTransaction, profile_id: str
+    ) -> dict:
+        """Deletes a Profile in a transaction."""
+        try:
+            p = await Profile.get_by_id(profile_id, transaction=transaction)
+            await p.delete(transaction=transaction)
+            return {"id": profile_id}
+        except ModelNotFoundError:
+            raise ValueError(f"Profile not found: {profile_id}")
+
+    p = Profile(name="Foo")
+    await p.save()
+    assert p.id
+
+    transaction = CONFIGURATIONS["db"].transaction()
+    result = await delete_in_transaction(transaction, p.id)
+    assert isinstance(result, dict)
+    assert result == {"id": p.id}
+
+    transaction2 = CONFIGURATIONS["db"].transaction()
+    with pytest.raises(ValueError) as excinfo:
+        await delete_in_transaction(transaction2, p.id)
+    assert str(excinfo.value) == f"Profile not found: {p.id}"
+
+
+@pytest.mark.asyncio
+async def test_update_model_in_transaction(configure_db) -> None:
+    """Test updating a model in a transaction."""
+
+    @async_transactional  # type: ignore
+    async def update_in_transaction(
+        transaction: AsyncTransaction, profile_id: str, name: str
+    ) -> Profile:
+        """Updates a Profile in a transaction."""
+        p = Profile(id=profile_id)
+        await p.reload(transaction=transaction)
+        p.name = name
+        await p.save(transaction=transaction)
+        return p
+
+    p = Profile(name="Foo")
+    await p.save()
+    assert p.id
+
+    transaction = CONFIGURATIONS["db"].transaction()
+    result = await update_in_transaction(transaction, p.id, "Bar")
+    assert isinstance(result, Profile)
+    assert result.id == p.id
+    assert result.name == "Bar"
+
+
+@pytest.mark.asyncio
+async def test_update_submodel_in_transaction(configure_db) -> None:
+    """Test Updating a submodel in a transaction."""
+
+    @async_transactional  # type: ignore
+    async def update_submodel_in_transaction(
+        transaction: AsyncTransaction, user_id: str, period: str, purchases: int
+    ) -> UserStats:
+        """Updates a UserStats in a transaction."""
+        u = await User.get_by_id(user_id)
+        us = UserStats.model_for(u)
+        user_stats: UserStats = await us.get_by_id(period)  # pylint: disable=no-member
```

Comment:
It might be nice if this `get_by_id` would also be passed the transaction. Since the test is not incrementing the value, rather just setting it to a fixed value it's not a big issue, but would be nice if the example would show correct use. Maybe this test could also be changed to increment the value instead of setting it to a fixed value to make it resemble real-life more.